packages/build/src/log/messages/compatibility.js

```diff
@@ -0,0 +1,173 @@
+'use strict'
+
+const { lt: ltVersion } = require('semver')
+
+const { isPreviousMajor } = require('../../utils/semver')
+const { getPluginOrigin } = require('../description')
+const { logArray, logSubHeader } = require('../logger')
+const { logWarningArray, logWarningSubHeader } = require('../logger')
+const { THEME } = require('../theme')
+
+const logLoadingPlugins = function (logs, pluginsOptions, debug) {
+  const loadingPlugins = pluginsOptions
+    .filter(isNotCorePlugin)
+    .map((pluginOptions) => getPluginDescription(pluginOptions, debug))
+
+  if (loadingPlugins.length === 0) {
+    return
+  }
+
+  logSubHeader(logs, 'Loading plugins')
+  logArray(logs, loadingPlugins)
+}
+
+// We only logs plugins explicitly enabled by users
+const isNotCorePlugin = function ({ origin }) {
+  return origin !== 'core'
+}
+
+const getPluginDescription = function (
+  {
+    packageName,
+    pluginPackageJson: { version },
+    loadedFrom,
+    origin,
+    pinnedVersion,
+    latestVersion,
+    expectedVersion,
+    compatibleVersion,
+  },
+  debug,
+) {
+  const versionedPackage = getVersionedPackage(packageName, version)
+  const pluginOrigin = getPluginOrigin(loadedFrom, origin)
+  const description = `${THEME.highlightWords(packageName)}${versionedPackage} ${pluginOrigin}`
+  if (!debug) {
+    return description
+  }
+
+  const versions = Object.entries({
+    pinned: pinnedVersion,
+    latest: latestVersion,
+    expected: expectedVersion,
+    compatible: compatibleVersion,
+  })
+    .filter(hasVersion)
+    .map(getVersionField)
+
+  if (versions.length === 0) {
+    return description
+  }
+
+  return `${description} (${versions.join(', ')})`
+}
+
+const hasVersion = function ([, version]) {
+  return version !== undefined
+}
+
+const getVersionField = function ([versionFieldName, version]) {
+  return `${versionFieldName} ${version}`
+}
+
+// Print a warning message when old versions plugins are used.
+// This can only happen when they are installed to `package.json`.
+const logOutdatedPlugins = function (logs, pluginsOptions) {
+  const outdatedPlugins = pluginsOptions.filter(hasOutdatedVersion).map(getOutdatedPlugin)
+
+  if (outdatedPlugins.length === 0) {
+    return
+  }
+
+  logWarningSubHeader(logs, 'Outdated plugins')
+  logWarningArray(logs, outdatedPlugins)
+}
+
+const hasOutdatedVersion = function ({ pluginPackageJson: { version }, latestVersion }) {
+  return version !== undefined && latestVersion !== undefined && ltVersion(version, latestVersion)
+}
+
+const getOutdatedPlugin = function ({
```

Comment:
`getOutdatedPlugin()` and `getOutdatedDescription()` are the only functions with changes. Everything else was just moved from the former file.